gui-js/libs/ui-components/src/lib/import-csv/import-csv.component.ts

```diff
@@ -181,6 +182,10 @@ export class ImportCsvComponent extends Zoomable implements OnInit, AfterViewIni
       }
     });
     document.onkeydown = this.onKeyDown;
+
+    this.electronService.minsky.dimensions.$properties().then(dims => {
+      this.existingDimensionNames = Object.entries(dims).map(d => d[0]);
+    });
```

Comment:
Wouldn't it be simpler to use the Object.keys() method here? That way, you could write something like:
`this.existingDimensionNames =(await this.electronService.minsky.dimensions.$properties()).keys();`

Reply:
Changing ngOnInit to an async method is not recommended practice. The code will not be awaited so the .then syntax is less misleading. I will use the Object.keys suggestion.

Reply:
Why is declaring ngOnInit async considered bad practice? IIUC, Angular will never await a call on that method, but an explicit call could. So where's the problem?

In any case, for our purposes, the .then syntax is equivalent, since this method is not called explicitly, so its just a matter of style.